apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/springwebmvc/SpringServiceNameInstrumentation.java

```diff
@@ -71,9 +71,19 @@
 
         @Advice.OnMethodExit(suppress = Throwable.class, inline = false)
         public static void afterInitPropertySources(@Advice.This WebApplicationContext applicationContext) {
-            if (applicationContext.getServletContext() != null) {
-                tracer.overrideServiceNameForClassLoader(applicationContext.getServletContext().getClassLoader(), applicationContext.getEnvironment().getProperty("spring.application.name"));
+
+            String appName = applicationContext.getEnvironment().getProperty("spring.application.name", "");
+
+            // fallback when application name isn't set through an environment property
+            if (appName.isEmpty()) {
+                appName = applicationContext.getApplicationName();
```

Comment:
Better guard against `null`. I've seen `servletContext.getContextPath()` return null on some application servers.

Reply:
I was also wondering should we make this work for older versions of Spring framework < 3.x